appserver/extras/embedded/shell/glassfish-embedded-static-shell/pom.xml

```diff
@@ -671,6 +671,18 @@
             <groupId>javax.enterprise</groupId>
             <artifactId>cdi-api</artifactId>
             <optional>true</optional>
+            <exclusions>
```

Comment:
Please explain the reasons for these 2 changes.

Reply:
otherwise classpath in MANIFEST.MF is generated with javax.inject.jar, but this jar is no longer in GF

Reply:
Hm, this one seems more legit than the other one that is similar.
Add the exclusion in dependencyManagement under appserver/pom.xml that will fix it for all modules.

Reply:
If it gets excluded, we need to add missing dependency org.glassfish.hk2.external:jakarta.inject to every pom where the cdi api is used. Javax.inject is a dependency in all of them

Reply:
@romain-grecourt do you want me to replace javax.inject with org.glassfish.hk2.external:jakarta.inject in all jars? There is about 6 of them

Reply:
As long as these modules use javax.inject in their code, this is fine. If so please do it.

Reply:
@romain-grecourt Actually, appserver/pom.xml for some reason has a dependency on javax.inject:javax.inject. Should we not replace it with  org.glassfish.hk2.external:jakarta.inject?

Reply:
That would solve the dependency everywhere

Reply:
This is dependency management right ? Yes we should change that.

Reply:
It does not work. Jakarta.inject.jar does not go to .m2, so it is not resolvable during a build. It is loaded only in GF itself, after it is build. So the dependency on javax.inject:javax.inject is needed.

Reply:
What does not work exactly ? This is managed dependency it does not trigger dependency resolution nor it adds a dependency to every module.

Why is it still needed ? If you add the new jakarta-inject dependency on the 6 modules you mentioned (as long as they actually depend on javax.inject in their code), what happens ?

Reply:
the new jakarta-inject dependency is not resolvable by maven. it is not in .m2

Reply:
If the javax.inject is excluded globally, then it is needed to be added to maven dependencies again, for compile, but it cannot be the new  org.glassfish.hk2.external:jakarta.inject.

Reply:
So basically it makes sense to exclude them, as then the glasspath is fixed, and the jakarta.inject.jar is put to classpath in MANIFEST.MF, but it does not make sense to add the dependency

Reply:
Where would maven take the jar from? It clearly cannot find it, when used as a dependency, the build fail. You can try it in your environment

Reply:
Saying that a dependency is not resolvable because it's not in the .m2 folder does not make sense. `org.glassfish.hk2.external:jakarta.inject:2.5.0` is available in OSSRH staging, see https://oss.sonatype.org/content/repositories/staging/org/glassfish/hk2/external/jakarta.inject/2.5.0/

Reply:
good point